import "job.proto";
//import "common.proto";

message EmbeddingProto
{
	optional int32 word_dim=1;
	optional int32 vocab_size=2;
	optional int32 batch_size=3;
}

message LossProto
{
	optional int32 nclass=1;
	optional int32 vocab_size=2;
}

message DataProto
{
	required string path=1;
	optional int32 max_window=2;
	optional string backend=3 [default="kvfile"];
	optional int32 batch_size=4;
}

message UnrollProto
{

	optional int32 batch_size=1;
	optional int32 max_window=2;

}

message HiddenProto
{
optional int32 word_dim=1;

optional int32 batch_size=2;
}

extend singa.LayerProto
{
	optional EmbeddingProto embedding_conf=101;
	optional LossProto loss_conf=102;
	optional DataProto data_conf=103;
	optional UnrollProto unroll_conf=104;
	optional HiddenProto hid_conf=105;
}

message WordRecord
{
	optional string word = 1;
	optional int32 word_index = 2;
  	optional int32 class_index = 3;
  	optional int32 class_start = 4;
  	optional int32 class_end = 5;
}
